internal/daemonset/daemonset.go

```diff
@@ -171,3 +172,44 @@ func getVolumes() []corev1.Volume {
 		},
 	}
 }
+
+func (d *daemonset) SetWorkerDaemonsetAsDesired(ctx context.Context, nfdInstance *nfdv1.NodeFeatureDiscovery, workerDS *appsv1.DaemonSet) error {
+	workerDS.ObjectMeta.Labels = map[string]string{"app": "nfd"}
+
+	workerDS.Spec = appsv1.DaemonSetSpec{
+		Selector: &metav1.LabelSelector{
+			MatchLabels: getWorkerLabelsAForApp("nfd-worker"),
+		},
+		Template: corev1.PodTemplateSpec{
+			ObjectMeta: metav1.ObjectMeta{
+				Labels: getWorkerLabelsAForApp("nfd-worker"),
+			},
+			Spec: corev1.PodSpec{
+				Tolerations: []corev1.Toleration{
+					{
+						Operator: "Exists",
+						Effect:   "NoSchedule",
+					},
+				},
+				Affinity: getWorkerAffinity(),
+
+				ServiceAccountName: "nfd-worker",
+				DNSPolicy:          corev1.DNSClusterFirstWithHostNet,
+				Containers: []corev1.Container{
+					{
+						Env:             getEnvs(),
+						Image:           nfdInstance.Spec.Operand.ImagePath(),
+						Name:            "nfd-worker",
+						Command:         []string{"nfd-worker"},
+						Args:            []string{"--server=nfd-master:$(NFD_MASTER_SERVICE_PORT)"},
```

Comment:
Just a note that this is only relevant if the NodeFeature API is disabled and gRPC is enabled (`--enable-nodefeature-api=false` in NFD v0.0.14 and v0.15.0). We could remove this/these especially if we're not planning to have a setting for enabling the (deprecated) gRPC API. WDYT?

```suggestion
						Args:            []string{},
```

It shouldn't hurt to have it, though (just ignored by default as NodeFeature API is enabled)

Reply:
I've left this as is for the moment.  I dont think we're intending to enable deprecated features, so I'm happy to remove it , unless @yevgeny-shnaidman disagrees?

Reply:
@chr15p lets remove it